apps/web/app/api/v1/management/attributes/[attributeId]/route.ts

```diff
@@ -0,0 +1,117 @@
+  if (!attribute) {
+    return null;
+  }
+  if (!(await canUserAccessAttribute(authentication, attribute))) {
+    throw new Error("Unauthorized");
+  }
+  return attribute;
+}
+
+const canUserAccessAttribute = async (authentication: any, attribute: TAttributeClass): Promise<boolean> => {
+  if (!authentication) return false;
+
+  if (authentication.type === "session") {
+    return await hasUserEnvironmentAccess(authentication.session.user, attribute.environmentId);
+  } else if (authentication.type === "apiKey") {
+    return attribute.environmentId === authentication.environmentId;
+  } else {
+    throw Error("Unknown authentication type");
+  }
+};
+
+export async function GET(
+  request: Request,
+  { params }: { params: { attributeId: string } }
+): Promise<NextResponse> {
+  try {
+    const authentication = await authenticateRequest(request);
+    const attribute = await fetchAndValidateAttribute(authentication, params.attributeId);
+    if (attribute) {
+      return responses.successResponse(attribute);
+    }
+    return responses.notFoundResponse("Attribute", params.attributeId);
+  } catch (error) {
+    return handleErrorResponse(error);
+  }
+}
+
+export async function DELETE(
+  request: Request,
+  { params }: { params: { attributeId: string } }
+): Promise<NextResponse> {
+  try {
+    const authentication = await authenticateRequest(request);
+    const attribute = await fetchAndValidateAttribute(authentication, params.attributeId);
+    if (!attribute) {
+      return responses.notFoundResponse("Survey", params.attributeId);
+    }
+    const deletedAttribute = await deleteAttributeClass(params.attributeId);
+    return responses.successResponse(deletedAttribute);
+  } catch (error) {
+    return handleErrorResponse(error);
+  }
+}
+
+export async function PUT(
+  request: Request,
+  { params }: { params: { attributeId: string } }
+): Promise<NextResponse> {
+  try {
+    const authentication = await authenticateRequest(request);
+    const attribute = await fetchAndValidateAttribute(authentication, params.attributeId);
+    if (!attribute) {
+      return responses.notFoundResponse("Survey", params.attributeId);
+    }
+    const attributeCLassUpdate = await request.json();
+    const inputValidation = ZAttributeClassUpdateInput.safeParse(attributeCLassUpdate);
+    if (!inputValidation.success) {
+      return responses.badRequestResponse(
+        "Fields are missing or incorrectly formatted",
+        transformErrorToDetails(inputValidation.error)
+      );
+    }
+    const updatedAttribute = await updatetAttributeClass(params.attributeId, inputValidation.data);
+    if (updatedAttribute) {
+      return responses.successResponse(updatedAttribute);
+    }
+    return responses.internalServerErrorResponse("Some error ocured while updating action");
+  } catch (error) {
+    return handleErrorResponse(error);
+  }
+}
+
+function handleErrorResponse(error: any): NextResponse {
```

Comment:
this function is used many times. Can we put this in it's own file, maybe in the`@formbricks/lib/services/auth` file?

Reply:
Can we name that file error.ts instead of auth ?

Reply:
Or if these are limited to api we can keep them in `app/api/v1/auth.ts`